operator/src/lib.rs

```diff
@@ -293,11 +297,17 @@ impl OpaState {
         ));
         container_builder.command(start_command);
         container_builder.add_configmapvolume(cm_name, "conf".to_string());
-
-        for env in env_vars {
-            if let Some(val) = env.value {
-                container_builder.add_env_var(env.name, val);
-            }
+        container_builder.add_env_vars(env_vars);
+
+        let mut annotations = BTreeMap::new();
+        // only add metrics container port and annotation if available
+        if let Some(metrics_port) = port {
+            annotations.insert(SHOULD_BE_SCRAPED.to_string(), "true".to_string());
+            container_builder.add_container_port(
+                ContainerPortBuilder::new(metrics_port.parse()?)
+                    .name("metrics")
+                    .build(),
+            );
```

Comment:
What about the "client" container port ?

Reply:
They are identical.